### PR TITLE
Fixes #2870 - Use the container IP for HCs only if an IP address was requested

### LINKED ISSUE #2870: Marathon healthchecks on wrong address
I have a basic app running in Docker, listening on 8080 in the container, forwarded to `31022`:

```
042d3effd90f        node:5.3.0          "/bin/sh -c 'node ${M"   2 minutes ago       Up 2 minutes        0.0.0.0:31022->8080/tcp   mesos-fa72a494-eb7a-4a0b-b927-6db16019a58b-S8.e8ca0809-f3b4-4bcc-8398-07b6628b110e
```

The marathon definition has a small health check:

```json
  "healthChecks": [
    {
        "protocol": "HTTP",
        "path": "/",
        "gracePeriodSeconds": 60,
        "intervalSeconds": 10,
        "portIndex": 0,
        "timeoutSeconds": 5,
        "maxConsecutiveFailures": 200
    }
  ]
```

Unfortunately, the healthchecks are not passing as I noticed Marathon is checking the wrong IP `172.17.0.2`:

```
Dec 23 01:43:49 mesos-master-i-1208acc8 marathon[15050]: [2015-12-23 01:43:49,683] WARN Connection attempt to 172.17.0.2:31304 failed in response to GET request to / with 1 retries left, retrying... (spray.can.client.HttpHostConnectionSlot:marathon-akka.actor.default-dispatcher-768)

Dec 23 01:43:49 mesos-master-i-1208acc8 marathon[15050]: [2015-12-23 01:43:49,683] WARN Connection attempt to 172.17.0.2:31304 failed in response to GET request to / with 4 retries left, retrying... (spray.can.client.HttpHostConnectionSlot:marathon-akka.actor.default-dispatcher-768)
```

Here is an `ifconfig` on Marathon host (same on Mesos slave):

```
docker0   Link encap:Ethernet  HWaddr 02:42:ef:33:e7:5d
          inet addr:172.17.0.1  Bcast:0.0.0.0  Mask:255.255.0.0
          UP BROADCAST MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:0 (0.0 B)  TX bytes:0 (0.0 B)

eth0      Link encap:Ethernet  HWaddr 0a:1b:a0:28:e2:63
          inet addr:10.92.42.112  Bcast:10.92.42.255  Mask:255.255.255.0
          inet6 addr: fe80::81b:a0ff:fe28:e263/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:9001  Metric:1
          RX packets:3188494 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3221752 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:1165507376 (1.1 GB)  TX bytes:710361280 (710.3 MB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:2077629 errors:0 dropped:0 overruns:0 frame:0
          TX packets:2077629 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0
          RX bytes:610406077 (610.4 MB)  TX bytes:610406077 (610.4 MB)
```

This `172.17.0.2` is the `eth0` on the actual Docker container and this is definitely wrong as it can't be reached from the actual host.

Any ideas?

### PR DESCRIPTION
**Pull request by gkleiman**

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/health/HealthCheckActor.scala [hunk 1]
```diff
@@ -87,11 +88,15 @@ class HealthCheckActor(
 
   def dispatchJobs(): Unit = {
     log.debug("Dispatching health check jobs to workers")
-    taskTracker.getTasks(appId).foreach { task =>
-      if (task.getVersion == appVersion && task.hasStartedAt) {
-        log.debug("Dispatching health check job for task [{}]", task.getId)
-        val worker: ActorRef = context.actorOf(workerProps)
-        worker ! HealthCheckJob(task, healthCheck)
+    appRepository.app(appId, Timestamp(appVersion)).foreach { maybeApp =>
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/health/HealthCheckActor.scala]: This actor is for one application/version and one health check.
Can you pass the appDefinition in the ctor?

**Comment by aquamatthias** [comment]
Please add a log statement (warning) in case the appId/version can not be found.

**Comment by gkleiman** [comment]
@aquamatthias thanks for the review! I hopefully addressed all your comments.

**Comment by aquamatthias** [comment]
LGTM - Thanks!

**Pull request merged**